frontend/internal-packages/pglite-server/src/extensionUtils.ts

```diff
@@ -0,0 +1,208 @@
+        return module.amcheck
+      }
+      case 'auto_explain': {
+        const module = await import('@electric-sql/pglite/contrib/auto_explain')
+        return module.auto_explain
+      }
+      case 'bloom': {
+        const module = await import('@electric-sql/pglite/contrib/bloom')
+        return module.bloom
+      }
+      case 'cube': {
+        const module = await import('@electric-sql/pglite/contrib/cube')
+        return module.cube
+      }
+      case 'earthdistance': {
+        const module = await import(
+          '@electric-sql/pglite/contrib/earthdistance'
+        )
+        return module.earthdistance
+      }
+      case 'fuzzystrmatch': {
+        const module = await import(
+          '@electric-sql/pglite/contrib/fuzzystrmatch'
+        )
+        return module.fuzzystrmatch
+      }
+      case 'isn': {
+        const module = await import('@electric-sql/pglite/contrib/isn')
+        return module.isn
+      }
+      case 'lo': {
+        const module = await import('@electric-sql/pglite/contrib/lo')
+        return module.lo
+      }
+      case 'seg': {
+        const module = await import('@electric-sql/pglite/contrib/seg')
+        return module.seg
+      }
+      case 'tablefunc': {
+        const module = await import('@electric-sql/pglite/contrib/tablefunc')
+        return module.tablefunc
+      }
+      case 'tcn': {
+        const module = await import('@electric-sql/pglite/contrib/tcn')
+        return module.tcn
+      }
+      case 'tsm_system_rows': {
+        const module = await import(
+          '@electric-sql/pglite/contrib/tsm_system_rows'
+        )
+        return module.tsm_system_rows
+      }
+      case 'tsm_system_time': {
+        const module = await import(
+          '@electric-sql/pglite/contrib/tsm_system_time'
+        )
+        return module.tsm_system_time
+      }
+      default:
+        return null
+    }
+  } catch (error) {
+    console.error(
+      `Failed to dynamically import extension ${extensionName}:`,
+      error,
+    )
+    return null
+  }
+}
+
+/**
+ * Load and filter extensions for PGlite
+ * Returns both the extension modules object and the list of supported extension names
+ */
+export async function loadExtensions(requiredExtensions: string[]): Promise<{
+  extensionModules: Extensions
+  supportedExtensionNames: string[]
+}> {
+  const extensionModules: Extensions = {}
+  const supportedExtensionNames: string[] = []
+
+  for (const ext of requiredExtensions) {
+    const normalizedExt = normalizeExtensionName(ext)
+
+    // Try to dynamically load the extension module
+    const extensionModule = await loadExtensionModule(normalizedExt)
+
+    if (extensionModule) {
+      try {
+        extensionModules[normalizedExt] = extensionModule
+        supportedExtensionNames.push(ext) // Add original extension name to supported list
+      } catch (error) {
+        console.error(`Failed to configure extension ${normalizedExt}:`, error)
+      }
```

Comment:
Why is this try-catch necessary? If you just want to push, I don't think it is necessary.

Reply:
You're absolutely right. Since it's unnecessary, I deleted it!
Thank you for noticing!
https://github.com/liam-hq/liam/pull/3270/commits/4511c098e1d24c5ede3dc8aa9ff74670db4a39c9